### PR TITLE
Basic file structure for quickstart-node application

### PR DESCRIPTION
**Pull request by AvnerMaster**
The structure:
Errors
Services - for the controllers/services/repositories the user wants to add
Utilities - Shared utilities between services
Types
Middlewares
 
I also added a basic README.md file

### CODE REVIEW

File: quickstart-node/src/services/index.ts
Comment (oricho123) [inline quickstart-node/src/services/index.ts:1]: Maybe prepare different folders for mcode-sdk services and "web services" (which I didn't do in the python quickstart)

Comment (oricho123) [inline quickstart-node/src/services/index.ts:1]: Also prepare routing folders\files :)

**Review by oricho123 — COMMENTED** [review]
💯 

After fixing the changes ask @gregra81 to review the file structure

File: quickstart-node/src/services/index.ts
Comment (AvnerMaster) [reply quickstart-node/src/services/index.ts:1]: The routing will be handled through the controllers, for example this structure:
src/services/auth/controller.ts
inside that file there will be routes relating to authentication.

If you have a better suggestion, feel free to tell me

Comment (AvnerMaster) [reply quickstart-node/src/services/index.ts:1]: Added

Comment (oricho123) [reply quickstart-node/src/services/index.ts:1]: Should it be under the services folder then? maybe change only the 'services' name

Comment (AvnerMaster) [reply quickstart-node/src/services/index.ts:1]: yes it should, and it is a service, so I don't think we should change the folder's name

Comment (AvnerMaster) [reply quickstart-node/src/services/index.ts:1]: The dilemma here is should we make the structure that for each service every file related to that service will be in its related folder. Or make it like "crosswise" like how you did in your python example, that each file related to auth for example will be in its related folder

Comment (oricho123) [reply quickstart-node/src/services/index.ts:1]: Yes, I get your dilemma; it depends on your project, I guess. If there are going to be "services" that need to be communicated with, then maybe they should be "shared" services and be treated differently.

And if some services are not shared but do need other service's functions then maybe we need to work with "managers" which combines services returns. 

but again this project shouldn't be THAT big so I'm not so sure if it's needed. whatever you'll choose will be fine. just note that the keyword "services" may be confusing for some and if they will contain controller.ts,serivce.ts,... then maybe a different name is better :)

Comment (AvnerMaster) [reply quickstart-node/src/services/index.ts:1]: I agree, lets make things simple, i will go with the routes folder approach, and the client can add controllers folder if he wants

**Pull request closed without merging**